common/types/hashable.go

```diff
@@ -0,0 +1,13 @@
+package types
+
+// A chain-agnostic generic interface to represent the following native types on various chains:
+// PublicKey, Address, Account, BlockHash, TxHash
+//
+//go:generate mockery --quiet --name Hashable --output ./mocks/ --case=underscore
+type Hashable[T any] interface {
```

Comment:
Could constrain to `Hashable`:
```suggestion
type Hashable[T Hashable] interface {
```

Reply:
Er maybe not that simple - seems fine as-is, if documented clearly.

Reply:
Doesn't allow me. Error "Use of generic type without instantiation"